### ISSUE TITLE
406 error when installing from github

### ISSUE THREAD

**Issue by smussell** [npm/npm#7560]
After updating to node 0.12.0 on mac OS 10.10.3 I started  getting a 406 error whenever I try to npm install from github.  My colleague also has this issue, so I don't believe it is an issue with my mac.  Here is a simplified package.json that exhibits the issue:

```
{
  "name": "test",
  "description": "test",
  "version": "0.0.1",
  "private": true,
  "keywords": "",
  "dependencies": {
    "react-grid-layout": "git+https://github.com/STRML/react-grid-layout.git"
    
  }
}
```

The output is:

```
npm ERR! fetch failed https://github.com/strml/react-draggable
npm WARN retry will retry, error on last attempt: Error: fetch failed with status code 406
npm ERR! fetch failed https://github.com/strml/react-draggable
npm WARN retry will retry, error on last attempt: Error: fetch failed with status code 406
npm ERR! fetch failed https://github.com/strml/react-draggable
npm ERR! Darwin 14.3.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install"
npm ERR! node v0.12.0
npm ERR! npm  v2.5.1

npm ERR! fetch failed with status code 406

```

This happens for all packages I've tried to install from github.

**Comment by smikes** [comment]
To rule out a node 0.12.0 bug, could you try using `io.js` 1.4.x or later?

The 406 code is not something that `npm` is generating or expects; it seems to be what `npm-registry-client` is getting back from interacting with the `request` module or with the underlying https code.  There is a known problem with https in `node@0.12` (<https://github.com/npm/npm/issues/7349>); this may be related.

**Comment by smussell** [comment]
I actually made some progress in troubleshooting, and its very strange.  Seems like it may not be an npm 12 issue.  Previously I had installed node 0.10.26 via [n](https://www.npmjs.com/package/n). Then I reinstalled node by doing brew uninstall and brew install now it works.  So my steps to reproduce are:

 1. brew install node
 2. npm install // works fine
 2. npm install -g n
 2. n 0.10.26
 3. npm install // works fine
 4. n 0.12.0
 5. npm install // I see the above error

What's strange is that if I replace step 4 with `n 0.10.36` all is fine.  Something about having previously installed .26 is causing the issue, it seems then this issue is probably an issue with n, so I'll close this.
